core/peer/peer.go

```diff
@@ -322,6 +322,14 @@ func (p *PeerImpl) GetPeers() (*pb.PeersMessage, error) {
 	return peersMessage, nil
 }
 
+func getPeerAddresses(peersMsg *pb.PeersMessage) []string {
+	var addresses []string
+	for _, v := range peersMsg.GetPeers() {
+		addresses = append(addresses, v.Address)
```

Comment:
Not certain how frequently this is called, but the `addresses` length looks like it could be computed to reduce allocations rather than appended up to size.